core/kazoo_json/src/kz_json.erl

```diff
@@ -255,11 +255,12 @@ from_list_recursive(L)
 recursive_from_list([First | _]=List)
   when is_list(List)
        andalso is_tuple(First)  ->
-    set_values([{K, recursive_from_list(V)} || {K,V} <- List], new());
+    set_values([{kz_term:to_binary(K), recursive_from_list(V)} || {K,V} <- List], new());
 recursive_from_list(X) when is_float(X) -> X;
 recursive_from_list(X) when is_integer(X) -> X;
 recursive_from_list(X) when is_atom(X) -> X;
-recursive_from_list(X) when is_list(X) -> X;
+recursive_from_list([X1 | _]=X) when is_list(X1) -> [kz_term:to_binary(Xn) || Xn <- X];
```

Comment:
this converts integers too?

I would really really love tests here. At least regression tests. It takes like 5 minutes to do...